themes/default/content/blog/kubernetes-architecture-templates/index.md

```diff
@@ -0,0 +1,388 @@
+---
+title: "Using Kubernetes Arch Templates with Poetry and Python"
+date: 2022-12-07
+meta_desc: Set up a Google Kubernetes Engine cluster for a web application with archtecture templates, all with Python and Poetry.
+meta_image: meta.png
+authors:
+    - laura-santamaria
+tags:
+    - kubernetes
+    - arch-templates
+    - templates
+---
+
+When building with Kubernetes for the first time, we often need to stand up a lot of infrastructure just to get to the point of having a base to build an application. Let's explore how we can wire together two of our architecture templates to generate a base for a web application running on Kubernetes on Google Cloud with Python and Poetry.
+
+<!--more-->
+
+## A primer on microservices
+
+To work with Kubernetes, you're generally following a microservices architecture. In short, here's how to think about a microservices architecture: Each logical component of your application is its own individual process acting as a closed box that defines what data must come in and what data is delivered on the other end---what's known as a _service_. Services communicate with one another via APIs or similar small interfaces, and each service addresses one part of your business logic inside your application.
+
+In a typical, very basic web application set up in a generic microservices architecture, you might have a data store, a backend, and a frontend (a user interface, or UI). In practice, you might have multiple backends and data stores to address multiple activities like logging in, running a shopping cart, delivering shipping updates, and more.
+
+To ensure all of these services work together, an orchestrator like Kubernetes ensures each service is running and healthy, is connected to the others, and runs as expected.
+
+## A typical Kubernetes architecture
+
+A Kubernetes _cluster_ is a collection of one or more machines (either virtual or bare-metal) called _nodes_ that host groups of _containers_, with each group containing multiple copies of a microservice. These groups of containers are called _pods_. And then, to connect and expose those pods to one another and the world, we add in _services_.
+
+For our system that we'll build today, we're going to have a cluster running on Google Kubernetes Engine (GKE). We'll create a namespace on that Kubernetes instance, which will hold all of our Kubernetes app resources. We'll have a _deployment_ that takes the configuration we need for NGINX (a _ConfigMap_) and spins up a pod holding the containers running that NGINX server. Then, we'll have a service that exposes the ports so we can access our NGINX server from anywhere.
+
+## Start with templates
+
+We're going to build our entire architecture starting with two templates: Our GKE template and our Helm app template. [Learn more about our new architecture templates.](/blog/intro-architecture-templates)
```

Comment:
Except we're not using the Helm app template, correct? We're using the web app template.

Reply:
...derp. Thanks; I had tried to use the Helm one, but it didn't like this attempt. Fixing now.